sdk/storage/storage-blob/review/storage-blob.api.ts

```diff
@@ -0,0 +1,1424 @@
+  metadata?: Metadata;
+  sourceModifiedAccessConditions?: Models.ModifiedAccessConditions;
+}
+
+// @public
+interface BlobUndeleteOptions {
+  abortSignal?: AbortSignalLike;
+  customerProvidedKey?: Models.CpkInfo;
+}
+
+// @public
+class BlockBlobClient extends BlobClient {
+  constructor(connectionString: string, containerName: string, blobName: string, options?: NewPipelineOptions);
+  // WARNING: The type "Models.BlockBlobCommitBlockListResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlockBlobCommitBlockListResponse" needs to be exported by the package (e.g. added to index.ts)
+  commitBlockList(blocks: string[], options?: BlockBlobCommitBlockListOptions): Promise<Models.BlockBlobCommitBlockListResponse>;
+  // WARNING: The type "Models.BlockListType" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlockBlobGetBlockListResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlockListType" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlockBlobGetBlockListResponse" needs to be exported by the package (e.g. added to index.ts)
+  getBlockList(listType: Models.BlockListType, options?: BlockBlobGetBlockListOptions): Promise<Models.BlockBlobGetBlockListResponse>;
+  // WARNING: The type "Models.BlockBlobStageBlockResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlockBlobStageBlockResponse" needs to be exported by the package (e.g. added to index.ts)
+  stageBlock(blockId: string, body: HttpRequestBody, contentLength: number, options?: BlockBlobStageBlockOptions): Promise<Models.BlockBlobStageBlockResponse>;
+  // WARNING: The type "Models.BlockBlobStageBlockFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlockBlobStageBlockFromURLResponse" needs to be exported by the package (e.g. added to index.ts)
+  stageBlockFromURL(blockId: string, sourceURL: string, offset?: number, count?: number, options?: BlockBlobStageBlockFromURLOptions): Promise<Models.BlockBlobStageBlockFromURLResponse>;
+  // WARNING: The type "Models.BlockBlobUploadResponse" needs to be exported by the package (e.g. added to index.ts)
+  // WARNING: The type "Models.BlockBlobUploadResponse" needs to be exported by the package (e.g. added to index.ts)
+  upload(body: HttpRequestBody, contentLength: number, options?: BlockBlobUploadOptions): Promise<Models.BlockBlobUploadResponse>;
+  uploadBrowserData(browserData: Blob | ArrayBuffer | ArrayBufferView, options?: UploadToBlockBlobOptions): Promise<BlobUploadCommonResponse>;
+  uploadFile(filePath: string, options?: UploadToBlockBlobOptions): Promise<BlobUploadCommonResponse>;
+  uploadStream(stream: Readable, bufferSize: number, maxBuffers: number, options?: UploadStreamToBlockBlobOptions): Promise<BlobUploadCommonResponse>;
```

Comment:
The docs for uploadStream need to be cleaned up a bit.

Can we provide defaults for bufferSize and maxBuffers? Since we're passing in a readable stream, it can grab highWaterMark from there, and maxBuffers can be Infinity or something?

Reply:
Documentation re: high water mark is possibly wrong. Buffer size is really blockSize. Consider defaulting 8MB. .Net also calls this maxTransferLength.

maxBuffers can default to 5. Consider renaming to maxConcurrency.